src/PostgREST/Response.hs

```diff
@@ -111,14 +111,19 @@ createResponse QualifiedIdentifier{..} MutateReadPlan{mrMutatePlan} ctxApiReques
           , toAppliedHeader <$> preferMissing
           ]
 
-    if preferRepresentation == Full then
-      response HTTP.status201 (headers ++ contentTypeHeaders ctxApiRequest) (LBS.fromStrict rsBody)
-    else
-      response HTTP.status201 headers mempty
+    case preferRepresentation of
+      Just Full -> response HTTP.status201 (addPrefToHeaders headers Full ++ contentTypeHeaders ctxApiRequest) (LBS.fromStrict rsBody)
+      Just None -> response HTTP.status201 (addPrefToHeaders headers None) mempty
+      Just HeadersOnly -> if null rsLocation
+                            then response HTTP.status201 headers mempty
```

Comment:
Codecov is warning us about this line not being tested. How about a test for this?

You can check the coverage locally:

```
$ nix-shell
$ postgrest-coverage
-- you might have to do postgrest-clean before 
```

Then start an http server for the coverage dir:
```
$ python3 -m http.server 9000 -d coverage/
```

(Now that I think of it, we could wrap the above in a `postgrest-coverage-serve` nix command)

Then navigate to `http://localhost:9000/postgrest-11.1.0-inplace/PostgREST.Response.hs.html`

Reply:
@wolfgangwalther Ahh, you're right. I didn't notice the `file:///` (I use `neovim-qt` and it didn't show as clickable there).

One improvement could be to make it more noticeable like on the `postgrest-with-postgresql-*` commands:

```
$ postgrest-with-postgresql-15 postgrest-run

postgrest-with-postgresql-15: You can connect with: psql 'postgres:///postgres?host=/run/user/1000/postgrest/postgrest-with-postgresql-15-A8D/socket' -U postgres
postgrest-with-postgresql-15: You can tail the logs with: tail -f /run/user/1000/postgrest/postgrest-with-postgresql-15-A8D/db.log
```

Reply:
@steve-chavez @wolfgangwalther  I think code coverage is not working for me locally. After running `postgrest-coverage` my `coverage/` dir is still empty. I guess maybe I am having some configuration issues. Any idea what could be the issue here?
This is the result of my coverage run:
```
FAILED test/io/test_cli.py::test_cli[dump config] - test_cli.PostgrestError
FAILED test/io/test_cli.py::test_cli[no config] - test_cli.PostgrestError
FAILED test/io/test_cli.py::test_expected_config[aliases.config] - test_cli.PostgrestError
FAILED test/io/test_cli.py::test_expected_config[boolean-string.config] - test_cli.PostgrestError
FAILED test/io/test_cli.py::test_expected_config[boolean-numeric.config] - test_cli.PostgrestError
FAILED test/io/test_cli.py::test_expected_config[types.config] - test_cli.PostgrestError
================== 6 failed, 119 passed, 2 skipped in 35.09s ===================
Temporary directory kept at: /run/user/1000/postgrest/postgrest-with-postgresql-15-i8m
Temporary directory kept at: /run/user/1000/postgrest/postgrest-coverage-Cnq
```

Reply:
@taimoorzaeem `postgrest-coverage` also runs io tests and spec tests. If any of those fail, then the coverage won't run. Does `potsgrest-test-io` give the same results? Also try rerunning the coverage test, sometimes io tests fail like that.

Reply:
Yeah, io tests are a bit flaky (see https://github.com/PostgREST/postgrest/issues/2864). On CI they work most of the time but locally they can fail. FWIW `postgrest-test-io` always succeeds for me locally, but not `postgrest-with-postgresql-* postgrest-test-io`.

Reply:
@laurenceisla Yes, for me `postgrest-test-io` gives the same result.

Reply:
@taimoorzaeem Can you post the results of `postgrest-test-io -vv`?  Also some ideas:

- Check the first io test that fails. If it's one that has a `time.sleep` inside, try increasing the sleep time.
- Isolate the io test that fails with `postgrest-test-io -k <name>`

There are many tests that rely on `time.sleep(0.1)`, that wait time could not be enough depending on the machine. (we might be able to improve this with a configurable global sleep.. maybe even autotune it later)

Reply:
@steve-chavez This is the result of `postgrest-test-io -vv`
```
=========================== short test summary info ===========================
FAILED test/io/test_cli.py::test_cli[dump config] - test_cli.PostgrestError
FAILED test/io/test_cli.py::test_cli[no config] - test_cli.PostgrestError
FAILED test/io/test_cli.py::test_expected_config[aliases.config] - test_cli.PostgrestError
FAILED test/io/test_cli.py::test_expected_config[boolean-string.config] - test_cli.PostgrestError
FAILED test/io/test_cli.py::test_expected_config[boolean-numeric.config] - test_cli.PostgrestError
FAILED test/io/test_cli.py::test_expected_config[types.config] - test_cli.PostgrestError
================== 6 failed, 119 passed, 2 skipped in 35.21s ==================
Temporary directory kept at: /run/user/1000/postgrest/postgrest-with-postgresql-15-aYl
```
All the 6 errors messages are almost same:
```
---------------------------- Captured stderr call -----------------------------
07/Aug/2023:11:06:27 +0500: An error ocurred when trying to query database settings for the config parameters
07/Aug/2023:11:06:27 +0500: {"code":"PGRST000","details":"connection to server on socket \"/run/postgresql/.s.PGSQL.5432\" failed: fe_sendauth: no password supplied\n","hint":null,"message":"Database connection error. Retrying the connection."}
07/Aug/2023:11:06:27 +0500: connection to server on socket "/run/postgresql/.s.PGSQL.5432" failed: fe_sendauth: no password supplied

postgrest: thread killed
```
For me, `test_cli.py` is failing and there is no `time.sleep` in that.

Reply:
@taimoorzaeem That's strange, it's trying to connect to postgres without a password somehow.

Please open a separate issue with that report.